From cf00642792f05f59a647977fe8fdf73cce4476f5 Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Mon, 30 Nov 2015 11:39:13 +0000
Subject: [PATCH 182/195] staging: etnaviv: pass MMU into
 etnaviv_iommu_unmap_gem()

Pass the iommu structure into etnaviv_iommu_unmap_gem() rather than
pulling it out of the etnaviv_vram_mapping structure.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_gem.c |  2 +-
 drivers/staging/etnaviv/etnaviv_mmu.c | 11 +++--------
 drivers/staging/etnaviv/etnaviv_mmu.h |  3 ++-
 3 files changed, 6 insertions(+), 10 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_gem.c b/drivers/staging/etnaviv/etnaviv_gem.c
index 8655e60..449298b 100644
--- a/drivers/staging/etnaviv/etnaviv_gem.c
+++ b/drivers/staging/etnaviv/etnaviv_gem.c
@@ -496,7 +496,7 @@ void etnaviv_gem_free_object(struct drm_gem_object *obj)
 	list_for_each_entry_safe(mapping, tmp, &etnaviv_obj->vram_list,
 				 obj_node) {
 		WARN_ON(mapping->use);
-		etnaviv_iommu_unmap_gem(mapping);
+		etnaviv_iommu_unmap_gem(mapping->mmu, mapping);
 	}
 
 	drm_gem_free_mmap_offset(obj);
diff --git a/drivers/staging/etnaviv/etnaviv_mmu.c b/drivers/staging/etnaviv/etnaviv_mmu.c
index 32a0799..0d46a26 100644
--- a/drivers/staging/etnaviv/etnaviv_mmu.c
+++ b/drivers/staging/etnaviv/etnaviv_mmu.c
@@ -192,7 +192,7 @@ int etnaviv_iommu_map_gem(struct etnaviv_iommu *mmu,
 
 		list_for_each_entry_safe(m, n, &list, scan_node) {
 			list_del_init(&m->scan_node);
-			etnaviv_iommu_unmap_gem(m);
+			etnaviv_iommu_unmap_gem(mmu, m);
 		}
 
 		/*
@@ -227,18 +227,13 @@ int etnaviv_iommu_map_gem(struct etnaviv_iommu *mmu,
 	return ret;
 }
 
-void etnaviv_iommu_unmap_gem(struct etnaviv_vram_mapping *mapping)
+void etnaviv_iommu_unmap_gem(struct etnaviv_iommu *mmu,
+	struct etnaviv_vram_mapping *mapping)
 {
-	struct etnaviv_iommu *mmu;
 	struct etnaviv_gem_object *etnaviv_obj;
 
-	if (!mapping)
-		return;
-
 	WARN_ON(mapping->use);
 
-	mmu = mapping->mmu;
-
 	/* If the vram node is on the mm, unmap and remove the node */
 	if (mapping->vram_node.mm == &mmu->mm) {
 		etnaviv_obj = mapping->object;
diff --git a/drivers/staging/etnaviv/etnaviv_mmu.h b/drivers/staging/etnaviv/etnaviv_mmu.h
index e4cf8b9..367c29e 100644
--- a/drivers/staging/etnaviv/etnaviv_mmu.h
+++ b/drivers/staging/etnaviv/etnaviv_mmu.h
@@ -57,7 +57,8 @@ int etnaviv_iommu_unmap(struct etnaviv_iommu *iommu, u32 iova,
 int etnaviv_iommu_map_gem(struct etnaviv_iommu *mmu,
 	struct etnaviv_gem_object *etnaviv_obj, u32 memory_base,
 	struct etnaviv_vram_mapping **mapping);
-void etnaviv_iommu_unmap_gem(struct etnaviv_vram_mapping *mapping);
+void etnaviv_iommu_unmap_gem(struct etnaviv_iommu *mmu,
+	struct etnaviv_vram_mapping *mapping);
 void etnaviv_iommu_destroy(struct etnaviv_iommu *iommu);
 
 size_t etnaviv_iommu_dump_size(struct etnaviv_iommu *iommu);
-- 
2.6.2

